source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm

```diff
@@ -131,41 +132,46 @@ - (UIView *)renderButtons:(ACRView *)rootView
+                        childview:childview
+                       featureReg:featureReg
+                         rootView:rootView
+                           inputs:inputs
+                        superview:superview
+                       hostConfig:config
+                 accumulatedWidth:accumulatedWidth
+                accumulatedHeight:accumulatedHeight
+                         maxWidth:maxWidth
+                        maxHeight:maxHeight];
+    }
 
-        UIButton *button = nil;
+    if (!secondary.empty()) {
+        ++renderedBtnNum;
 
-        @try {
-            if ([acoElem meetsRequirements:featureReg] == NO) {
-                @throw [ACOFallbackException fallbackException];
-            }
-            button = [actionRenderer renderButton:rootView
-                                           inputs:inputs
-                                        superview:superview
-                                baseActionElement:acoElem
-                                       hostConfig:config];
-            [childview addArrangedSubview:button];
-        } @catch (ACOFallbackException *exception) {
-            handleActionFallbackException(exception, superview, rootView, inputs, acoElem, config,
-                                          childview);
-            NSUInteger count = [childview.arrangedSubviews count];
-            if (count > numElem) {
-                UIView *view = [childview.arrangedSubviews lastObject];
-                if (view && [view isKindOfClass:[UIButton class]]) {
-                    button = (UIButton *)view;
-                }
-            }
-        }
+        ACRBaseActionElementRenderer *actionRenderer =
+            [reg getActionRenderer:[NSNumber numberWithInt:(int)ActionType::Overflow]];
+
+        ACOBaseActionElement *overflow = [[ACOActionOverflow alloc] initWithBaseActionElements:secondary];
 
-        accumulatedWidth += [button intrinsicContentSize].width;
-        accumulatedHeight += [button intrinsicContentSize].height;
-        maxWidth = MAX(maxWidth, [button intrinsicContentSize].width);
-        maxHeight = MAX(maxHeight, [button intrinsicContentSize].height);
+        [self renderButtonForElem:overflow
+                   actionRenderer:actionRenderer
+                        childview:childview
+                       featureReg:featureReg
+                         rootView:rootView
+                           inputs:inputs
+                        superview:superview
+                       hostConfig:config
+                 accumulatedWidth:accumulatedWidth
+                accumulatedHeight:accumulatedHeight
+                         maxWidth:maxWidth
+                        maxHeight:maxHeight];
     }
 
     float contentWidth = accumulatedWidth;
+    
     if (ActionsOrientation::Horizontal == adaptiveActionConfig.actionsOrientation) {
-        contentWidth += (elems.size() - 1) * spacing;
+        contentWidth += (renderedBtnNum - 1) * spacing;
```

Comment:
i don't see renderedBtnNum is incremented for primary buttons.

Reply:
Check out the code right above:
```
    if (!secondary.empty()) {
        ++renderedBtnNum;
        // ....
    }
```